lib/katello/tasks/clean_backend_objects.rake

```diff
@@ -72,6 +72,14 @@ namespace :katello do
       end
     end
 
+    def host_unregister_options(host)
+      if host.managed? || host.compute_resource
```

Comment:
I would log "skipping Foreman record deletion for host <host.id>; host is managed by Foreman" or something similar.